raysect/core/math/function/function1d/cmath.pyx

```diff
@@ -147,3 +147,29 @@ cdef class Atan4Q1D(Function1D):
 
     cdef double evaluate(self, double x) except? -1e999:
         return cmath.atan2(self._numerator.evaluate(x), self._denominator.evaluate(x))
+
+cdef class Sqrt1D(Function1D):
+    """
+    A Function1D class that implements the square root of the result of a Function1D object: atan(f())
+
+    :param Function1D function: A Function1D object.
+    """
+    def __init__(self, object function):
+
+        self._function = autowrap_function1d(function)
+
+    cdef double evaluate(self, double x) except? -1e999:
+        return cmath.sqrt(self._function.evaluate(x))
```

Comment:
If the evaluation is -ve this needs to raise a ValueError exception as the function framework doesn't support complex numbers. The test will need to be update to confirm this.

Reply:
Added value check. In case of negative number ValueError is raised.